sqlcapture/capture.go

```diff
@@ -395,11 +402,21 @@ func (c *Capture) updateState(ctx context.Context) error {
 
 	// Likewise streams may be removed from the catalog, and we need to forget
 	// the corresponding state information.
-	for streamID, state := range c.State.Streams {
-		var _, streamExistsInCatalog = c.Bindings[streamID]
-		if !streamExistsInCatalog && state.Mode != TableModeIgnore {
-			logrus.WithField("stream", streamID).Info("stream removed from catalog")
-			c.State.Streams[streamID] = &TableState{Mode: TableModeIgnore, dirty: true}
+	// TODO(whb): We can eventually remove this when backfill counters are fully incorporated into
```

Comment:
Just checking my understanding here: The plan is that "backfill counter" will be incremented if the user disables+re-enables a single binding, even if they don't use the new "go re-backfill this" flow?

(Because in addition to permitting users to re-backfill a collection by toggling the binding, it's also important that this logic here currently _prevents_ the possibility of disabling a stream, missing some amount of replication events, and then re-enabling it _without_ triggering that new backfill.)

Reply:
That's a good point, we'll probably need to keep this state-clearing logic around since its not really valid to disable + resume a binding without re-backfilling it; I always seem to forget about that. I don't know that we'd always want to increment the backfill counter on re-enabling a binding, and although that might be possible, I don't think we'd do it right away 🤔.